### PR TITLE
feat: add exposed secret crd

### PR DESCRIPTION
**Pull request by josedonizetti**
Signed-off-by: Jose Donizetti <jdbjunior@gmail.com>

## Description

Adds ExposedSecrets support on container images. To avoid recreating jobs for the same workload, the PR uses the existing vulnerability controller implementation to also support exposed secrets. The implementation is working, tested and documented. Though I think we should refactor this in a following PR (this one is huge already), to extract the `vulnerabilityreport/controller` to `workload/controller`, the code would be the same, but it would be more meaningful to have a `workloadcontroller` who creates reports for both `vulnerability` and `exposedsecrets` using `pkg/vulnerabilityreport` and `pkg/exposedsecretreport.  Also, I chose not to support TTL for exposedsecrets on this PR and will implement it on a differrent PR. 

## Related issues
- Close https://github.com/aquasecurity/trivy-operator/issues/64

Remove this section if you don't have related PRs.

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)

### CODE REVIEW

File: pkg/exposedsecretreport/builder.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	b.hash = hash
+	return b
+}
+
+func (b *ReportBuilder) Data(data v1alpha1.ExposedSecretReportData) *ReportBuilder {
+	b.data = data
+	return b
+}
+
+func (b *ReportBuilder) ReportTTL(ttl *time.Duration) *ReportBuilder {
```
Comment (chen-keinan) [inline pkg/exposedsecretreport/builder.go:54]: Not sure if severity `unknown` is relevant to secret world. its more related to vulnerability scanning when it comes from an external DB and its, in secret we define the severity so unknown not likely to happen

File: deploy/crd/exposedsecretreports.crd.yaml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+          type: integer
+          name: Medium
+          description: The number of medium exposed secrets
+          priority: 1
+        - jsonPath: .report.summary.lowCount
+          type: integer
+          name: Low
+          description: The number of low exposed secrets
+          priority: 1
+        - jsonPath: .report.summary.unknownCount
```
Comment (chen-keinan) [inline deploy/crd/exposedsecretreports.crd.yaml:216]: Not sure if severity `unknown` is relevant to secret world. its more related vulnerability scaninng  when it comes from an external DB and its, in secret we define the severity so unknow not likely to happen

File: pkg/plugin/trivy/plugin.go [hunk 3]
```diff
@@ -1318,6 +1346,25 @@ func (p *plugin) toSummary(vulnerabilities []v1alpha1.Vulnerability) v1alpha1.Vu
...
+		case v1alpha1.SeverityCritical:
+			s.CriticalCount++
+		case v1alpha1.SeverityHigh:
+			s.HighCount++
+		case v1alpha1.SeverityMedium:
+			s.MediumCount++
+		case v1alpha1.SeverityLow:
+			s.LowCount++
+		default:
+			s.UnknownCount++
```
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:1362]: Not sure if severity `unknown` is relevant to secret world. its more related to vulnerability scanning when it comes from an external DB and its, in secret we define the severity so unknown not likely to happen

File: deploy/crd/exposedsecretreports.crd.yaml [hunk 4]
```diff
@@ -0,0 +1,229 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: exposedsecretreports.aquasecurity.github.io
+  labels:
+    app.kubernetes.io/managed-by: trivy-operator
+    app.kubernetes.io/version: "0.0.5"
```
Comment (chen-keinan) [inline deploy/crd/exposedsecretreports.crd.yaml:8]:
```suggestion
    app.kubernetes.io/version: "0.0.7"
```

File: pkg/apis/aquasecurity/v1alpha1/exposed_secrets_types.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	HighCount int `json:"highCount"`
+
+	// MediumCount is the number of exposed secrets with Medium Severity.
+	MediumCount int `json:"mediumCount"`
+
+	// LowCount is the number of exposed secrets with Low Severity.
+	LowCount int `json:"lowCount"`
+
+	// UnknownCount is the number of exposed secrets with unknown severity.
+	UnknownCount int `json:"unknownCount"`
```
Comment (chen-keinan) [inline pkg/apis/aquasecurity/v1alpha1/exposed_secrets_types.go:29]: see comment on `Unknown`

File: pkg/plugin/trivy/plugin.go (hunk 3 above)
Comment (chen-keinan) [inline pkg/plugin/trivy/plugin.go:1362]: same thing on unknown

**Review by chen-keinan — APPROVED** [review]

**Comment by chen-keinan** [comment]
@josedonizetti looking good , added few small commments

File: pkg/exposedsecretreport/builder.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+	report := v1alpha1.ExposedSecretReport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.reportName(),
+			Namespace: b.controller.GetNamespace(),
+			Labels:    labels,
+		},
+		Report: b.data,
+	}
+
+	if b.reportTTL != nil {
```
[the remark this answers is not in the document]
Comment (josedonizetti) [reply pkg/exposedsecretreport/builder.go:88]: This makes a lot of sense, doesn't seem applicable for secrets.

**Pull request merged**